bpm/bonita-core/bonita-process-engine/src/main/java/org/bonitasoft/engine/api/impl/ServerAPIImpl.java

```diff
@@ -470,13 +465,20 @@ protected Object invokeAPI(final Object[] parametersValues, final Object apiImpl
         }
     }
 
+    private void warnIfDeprecated(Method method, String apiInterfaceName) {
+        if (technicalLogger != null && method.isAnnotationPresent(Deprecated.class)) {
+            technicalLogger.log(this.getClass(), TechnicalLogSeverity.WARNING, "The API method " + apiInterfaceName + "." + method.getName()
+                    + " is deprecated. It will no longer exists in future major release. Check the javadoc to know how to replace it.");
```

Comment:
The API method ... is deprecated. It will be deleted in a future release. Please plan to update your code to use the replacement method instead. Check the Javadoc for more details.